### PR TITLE
Esbuild 0.17.x compatibility, bumped package versions, target latest

### PR DESCRIPTION
**Pull request by jonluca**
With electron 24 out, and esbuilt 0.17.x having its 17th released, figured I'd update this to work on the latest version

Targets ESNext, Node 19, and uses pnpm 8. Fixes up the watching and incremental logic to work with the new context in esbuilt 0.17.x

### CODE REVIEW

File: packages/create-app/package.json [hunk 1]
```diff
@@ -22,11 +22,11 @@
   },
   "exports": "./dist/index.mjs",
   "engines": {
-    "node": "^14.18 || >=16.0.0"
+    "node": ">=19"
```
Comment (Kiyozz) [inline packages/create-app/package.json]: Thanks for your efforts to improve electron-esbuild!

Meanwhile, I don't think the “engines” section should be upgraded. This should indicate which versions of node this package works for.

I’m not into using non-lts versions of node for libraries: please keep it on the minimum version Electron is supporting. This applies for rebuild config of templates too.

File: packages/create-app/package.json [hunk 2]
```diff
@@ -22,11 +22,11 @@
...
   },
   "scripts": {
-    "build": "ee 'src/**' --format esm --sourcemap --clean-outdir --outdir dist --target node16",
-    "prepublishOnly": "ee 'src/**' --format esm --sourcemap --clean-outdir --outdir dist --target node16 --check-types --ts-project tsconfig.types.json",
+    "build": "ee 'src/**' --format esm --sourcemap --clean-outdir --outdir dist --target node19",
+    "prepublishOnly": "ee 'src/**' --format esm --sourcemap --clean-outdir --outdir dist --target node19 --check-types --ts-project tsconfig.types.json",
```
Comment (Kiyozz) [inline packages/create-app/package.json]: See my comment for versions.

File: packages/create-app/templates/main-ts/esbuild.main.config.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ const config: BuildOptions = {
     path.resolve('src/main/preload.ts'),
   ],
   bundle: true,
-  target: 'node16.15.0', // electron version target
+  target: 'node19', // electron version target
```
Comment (Kiyozz) [inline packages/create-app/templates/main-ts/esbuild.main.config.ts]: See my comments for versions.

File: packages/create-app/tsconfig.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "dist",
     "lib": ["esnext"],
-    "target": "es2021",
+    "target": "ESNext",
```
Comment (Kiyozz) [inline packages/create-app/tsconfig.json]: Are you sure about this? Why this change?

File: packages/ee/package.json [hunk 5]
```diff
@@ -19,8 +19,8 @@
   },
   "exports": "./dist/index.mjs",
   "engines": {
-    "node": "^14.18 || >=16.0.0",
-    "pnpm": "7"
+    "node": ">=19",
+    "pnpm": ">=7"
```
Comment (Kiyozz) [inline packages/ee/package.json]: Then, this should be 8 here

File: packages/electron-esbuild/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "electron-esbuild",
-  "version": "7.0.0",
+  "name": "electron-esbuild-17x",
```
Comment (Kiyozz) [inline packages/electron-esbuild/package.json]: Do not change the package name

**Review by Kiyozz — CHANGES_REQUESTED** [review]

**Comment by jonluca** [comment]
Cool - yeah this was partially wanting to contribute up stream and partially trying to improve performance of my app, so removing polyfills/modernizing build targets made sense. Will remove those changes for the PR though - just updated

File: package.json [hunk 7]
```diff
@@ -11,22 +11,22 @@
   "private": true,
   "engines": {
     "node": ">=14.18",
-    "pnpm": "7"
+    "pnpm": ">=8"
```
Comment (Kiyozz) [inline package.json]:
```suggestion
    "pnpm": "8"
```

**Review by Kiyozz — CHANGES_REQUESTED** [review]
I finished review the code. When I can, I will test this! 🙂

**Review by Kiyozz — APPROVED** [review]

**Comment by Kiyozz** [comment]
I wanted to take a moment to express my sincere gratitude for your valuable contribution to electron-esbuild. Your recent support and pull-request proposal to fix the bug has been immensely appreciated. Once again, thank you for extending your support to electron-esbuild.

**Pull request merged**